src/EnergyPlus/SurfaceGeometry.cc

```diff
@@ -9615,7 +9615,10 @@ namespace SurfaceGeometry {
                                                                      "ONIFHIGHOUTDOORAIRTEMPANDHIGHSOLARONWINDOW",
                                                                      "ONIFHIGHOUTDOORAIRTEMPANDHIGHHORIZONTALSOLAR",
                                                                      "ONIFHIGHZONEAIRTEMPANDHIGHSOLARONWINDOW",
-                                                                     "ONIFHIGHZONEAIRTEMPANDHIGHHORIZONTALSOLAR"});
+                                                                     "ONIFHIGHZONEAIRTEMPANDHIGHHORIZONTALSOLAR",
+                                                                     "ONIFHIGHLUMINANCEORHIGHSOLARTILLMIDNIGHT",
+                                                                     "ONIFHIGHLUMINANCEORHIGHSOLARTILLSUNSET",
+                                                                     "ONIFHIGHLUMINANCEORHIGHSOLARTILLNEXTMORNING"});
```

Comment:
adding new shading control options

Reply:
We are replacing these `Array1D_string` with `constexpr std::array<std::string_view, static_cast<int>(X::Num)> XNamesUC = {};`. These arrays are compile time structures that are just loaded from the executable data section rather than constructed on the heap.

Reply:
Thanks! getEnumerationValue fixed the problem.